src/components/input/demoErrorsAdvanced/index.html

```diff
@@ -0,0 +1,95 @@
+<div ng-controller="AppCtrl" layout="column" ng-cloak>
+
+  <md-content layout-padding>
+    <form name="userForm">
+      <div layout="row" layout-xs="column" layout-sm="column" layout-align="space-between center">
+        <div flex-gt-sm="80">
+          <p>
+            The <code>md-input-container</code> gives you the flexibility to display your messages
+            using many standard angular directives.
+          </p>
+
+          <p>
+            For instance, toggle the checkbox
+
+            <span hide-xs hide-sm>to the right</span>
+            <span hide-gt-sm>below</span>
+
+            to see the messages switch between some custom hints, and the actual error messages.
+            Note that some of the <code>ng-messages</code> containers use <code>ngIf</code> while
+            others use <code>ng-show</code> or <code>ng-hide</code>.
+          </p>
+        </div>
+
+        <md-input-container>
+          <md-checkbox ng-model="showHints">Showing {{showHints ? "Hints" : "Errors"}}</md-checkbox>
```

Comment:
Ah, indeed! I was thinking of using a radio button, but a switch definitely works better.